File: src/main/java/com/venus/meetspace/dto/request/ActivitySearchRequest.java
package com.venus.meetspace.dto.request;

import lombok.Data;

import java.time.LocalDateTime;

@Data
public class ActivitySearchRequest {
    private LocalDateTime startTime;
    private LocalDateTime endTime;
    private double longitude; // 经度
    private double latitude; // 纬度
    private double radiusKm;
    private int min;
    private int max; // 参与人数范围
}

File: frontend/lib/model/request/activity_search_request.dart
import 'package:json_annotation/json_annotation.dart';

part 'activity_search_request.g.dart';

@JsonSerializable()
class ActivitySearchRequest {
  final DateTime? startTime;
  final DateTime? endTime;
  final double? longitude;
  final double? latitude;
  final double? radiusKm;
  final int? min;
  final int? max;

  ActivitySearchRequest({
    this.startTime,
    this.endTime,
    this.longitude,
    this.latitude,
    this.radiusKm,
    this.min,
    this.max,
  });

  factory ActivitySearchRequest.fromJson(Map<String, dynamic> json) =>
      _$ActivitySearchRequestFromJson(json);
  Map<String, dynamic> toJson() => _$ActivitySearchRequestToJson(this);
}
